### PR TITLE
SPCTR-3234 Improvement: Forms - If the  field is empty in form submissions, the default recipient will be the current user's WordPress email address.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-3134

### Screenshots

- https://d.pr/v/Bp5pZd

### Types of changes

- Improvements

### How has this been tested?

- Added to the card

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -221,6 +221,7 @@ Our external packages use [Rating Star Component](https://github.com/n49/react-s
 * Fix: Image Gallery - Resolved an issue with the masonry type wherein the loader would not be hidden after all the images have been loaded.
 * Fix: Global - Resolved an issue in which the font-family that was set would not appear as intended in the FSE template editor.
 * Fix: Separator - Resolved an issue wherein the height would revert from percentage to pixels after reloading the editor.
+* Improvement: Forms - If the `To` field is empty in form submissions, the default recipient will be the current user's WordPress email address.
```
Comment (bsf-zanev) [inline README.md:224]: Can we move this improvement up with the others?

```suggestion
* Improvement: Form - Made the default recipient as the current user's WordPress email address when the `To` field is left empty in submitted forms.
```

File: includes/blocks/forms/frontend.js.php [hunk 2]
```diff
@@ -9,15 +9,18 @@
 
 $selector        = '.uagb-block-' . $id;
 $current_post_id = get_the_ID();
-$js_attr         = array(
+$curr_user       = wp_get_current_user();
+$default_email   = $curr_user->user_email;
```
Comment (bsf-zanev) [inline includes/blocks/forms/frontend.js.php:13]: Is there a chance of gettin 0 here for `$curr_user`?

Should we add something like this?

```suggestion
$default_email   = ( $curr_user instanceof WP_User ) ? $curr_user->user_email : 'someplaceholder@e.mail';
```

### - OR -

```suggestion
$default_email   = ( $curr_user instanceof WP_User ) ? $curr_user->user_email : null;
```
And if null then handle it.

**Review by bsf-zanev — COMMENTED** [review]
Had one minor doubt.

File: README.md (hunk 1 above)
Comment (AbhijitNage123) [reply README.md]: Done

File: includes/blocks/forms/frontend.js.php (hunk 2 above)
Comment (AbhijitNage123) [reply includes/blocks/forms/frontend.js.php:13]: @bsf-zanev `$curr_user` we did not get 0 in any case and always a instance of WP_User so no need to check any condition

**Review by bsf-zanev — APPROVED** [review]

File: includes/blocks/forms/frontend.js.php [hunk 3]
```diff
@@ -9,15 +9,18 @@
...
+$js_attr = array(
 	'block_id'                => $attr['block_id'],
 	'reCaptchaEnable'         => $attr['reCaptchaEnable'],
 	'reCaptchaType'           => $attr['reCaptchaType'],
 	'reCaptchaSiteKeyV2'      => $attr['reCaptchaSiteKeyV2'],
 	'reCaptchaSecretKeyV2'    => $attr['reCaptchaSecretKeyV2'],
 	'reCaptchaSiteKeyV3'      => $attr['reCaptchaSiteKeyV3'],
 	'reCaptchaSecretKeyV3'    => $attr['reCaptchaSecretKeyV3'],
-	'afterSubmitToEmail'      => $attr['afterSubmitToEmail'],
+	'afterSubmitToEmail'      => ! empty( $attr['afterSubmitToEmail'] ) ? $attr['afterSubmitToEmail'] : $default_email,
```
Comment (adi3890) [inline includes/blocks/forms/frontend.js.php:23]: @AbhijitNage123 have we checked the scenario where spaces are entered.
Is that managed.
to manae that we can use.
trim()

Comment (adi3890) [reply includes/blocks/forms/frontend.js.php:23]: Seems like there are some issue.
Please confirm once.

Comment (AbhijitNage123) [reply includes/blocks/forms/frontend.js.php]: Done and retested

**Review by adi3890 — APPROVED** [review]

**Pull request merged**